### ISSUE TITLE
[A11ySev2] [Programmatic Access-Dashboards with Grafana –Azure Insights Key Vaults]:Ensure every form element has a label.

### ISSUE THREAD

**Issue by LinglingTong** [grafana/grafana#118767]
## Steps to reproduce

Repro Steps: 
1. Open 'Azure Monitor dashboards with Grafana' in Azure Portal
2. "Monitor | Dashboards with Grafana | Gallery" screen appears with already Dashboards with Grafana tab selected.  
3. Select already existing data as 'Azure Insights Key Vaults'  under All Dashboards tab.  
4. Navigate all control under Newley opened page like Tool bar , Controls under tool bar (dataSource , Subscription , Resource group,...etc) and overview section. 
5. Run Accessibility Insight Tool and Verify whether ARIA Combo box have an accessible name under Key Vaults tab or not. 

**Actual Result:**
The combobox element does not have an accessible name or label. Screen readers do not provide any meaningful information about the combobox, making it unclear to users what the purpose of the element is.  
**Observation**:  
The input element with the role of combobox lacks any attributes (aria-label, aria-labelledby, title, or placeholder) that would provide a programmatically defined label. This results in the combobox being announced without context, which can confuse users of assistive technologies.  
**Issue **1: Same issue repro for the table column header fields under the dimensions table under the 'Explore' page.  
**Navigation**: Open https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  >Select Azure Insights Key Vault> Select Edit>Invoke Add visualization> Invoke Back to dashboard > Navigate options under three dot menu button on new panel> Invoke Explore> Select Add dimension button >  Run Ai tool and verify the issue.  
**Issue 2**: Same issue repro for the Row options dialog box  after activating the settings button from the row title line.  
**Navigation**: Open https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  > Select New button button > Invoke New dashboard control> Select “ Add visualization” button > Select testdata list item> Select Back to dashboard in the tool bar > Select Add row from add menu button in the tool bar > Select setting button in the row title line > Run ai tool and verify the issue.  

**Expected Result:**
Every form element, including comboboxes, must have a programmatically defined accessible label to ensure that users understand its purpose. This can be achieved by associating the combobox with a descriptive <label> element using the for attribute or by adding an aria-label attribute with a meaningful description .  
 

 
 
**Issue**:  
 
Ensure every form element has a label (label - https://accessibilityinsights.io/info-examples/web/label)  
**Target application: ** 
 
Monitor - Microsoft Azure - https://portal.azure.com/#view/Microsoft_Azure_Monitoring/AzureMonitoringBrowseBlade/~/AzureGrafanaReactView  
**

**Relevant WCAG Criteria:** [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) (Level A)

## Environment

- **Grafana version:** 12.2.0
- **Data source type & version:**
- **User OS & Browser:** Version 25H2 (OS Build 26200.7093) / Chrome Version 142.0.7444.177 (Official Build) (64-bit)
- **Others:**

**Comment by LinglingTong** [comment]
<img width="2969" height="955" alt="Image" src="https://github.com/user-attachments/assets/1042b225-2141-442a-9d6c-1b4502785b75" />
<img width="2806" height="957" alt="Image" src="https://github.com/user-attachments/assets/5fbb02ce-00ca-4fde-a4cd-5d2d0fc0de2b" />
<img width="2651" height="953" alt="Image" src="https://github.com/user-attachments/assets/c687c9b7-72ea-40c4-99fe-ecc35ae2e2c5" />

**Comment by ashharrison90** [comment]
the first screenshot (row options) is fixed with the new dynamic dashboards

the other 2 are still valid, rerouting to dashboards + partner datasources respectively

**Comment by mdvictor** [comment]
regarding variables it was also fixed with https://github.com/grafana/scenes/pull/1404 and available in grafana 13

**Comment by joshhunt** [comment]
- [x] Confirmed Row repeat/row options is fixed in legacy dashboards
- [x] https://github.com/grafana/grafana/pull/130761 fixes Azure Monitor dimensions
- [x] Confirmed Dashboard variables fixed, including when dynamic dashboards is disabled as well
